src/modules/CRUD/components/Details/PropertyItem/Edit/Edit.js

```diff
@@ -0,0 +1,168 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from '@blueprintjs/core';
+import Form from 'react-jsonschema-form';
+import RTEField from '../../../../../../components/react-json-schemaForm/RTEField';
+import GeometryField from '../../../../../../components/react-json-schemaForm/GeometryField';
+import FileWidget from '../../../../../../components/react-json-schemaForm/FileWidget';
+import './styles.scss';
+
+const widgets = {
+  FileWidget: props => (
+    <FileWidget {...props} />
+  ),
+};
+
+const sanitizeValue = (type, value) => {
+  if (type === 'array' && value === null) {
+    return [];
+  }
+  return value;
+};
+
+const Edit = ({
+  editedItem,
+  name,
+  onEdit,
+  onRead,
+  setEditedItem,
+  schema,
+  schema: { type: schemaType },
+  t,
+  ui_schema: uiSchema,
+  value,
+  ...rest
+}) => {
+  const isCurrentEditedItem = editedItem === name;
+  const canEdit =  ['', name].includes(editedItem);
+  const [loading, setLoading] = useState(false);
+  const [defaultValue, setDefaultValue] = useState(sanitizeValue(schemaType, value));
+
+  useEffect(() => {
+    isCurrentEditedItem && !loading ? onEdit(name) : onRead(name);
+  }, [isCurrentEditedItem, loading, name, onEdit, onRead]);
+
+  useEffect(() => {
+    setDefaultValue(sanitizeValue(schemaType, value));
+  }, [schemaType, value]);
+
+  const submitFeature = async ({ formData }) => {
```

Comment:
I'm not sure about this one; this function is called after the form is submitted. I think to put this function outside the component to avoid unnecessary calculations when rendering.